src/transactions/TransactionUtils.cpp

```diff
@@ -1159,6 +1174,275 @@ removeOffersByAccountAndAsset(AbstractLedgerTxn& ltx, AccountID const& account,
+
+    prefetchForRevokeFromPoolShareTrustLines(ltxInner, poolShareTrustLines);
+
+    for (auto& poolShareTrustLine : poolShareTrustLines)
+    {
+        // use a lambda so we don't hold a reference to the internals of
+        // TrustLineEntry
+        auto poolTL = [&]() -> TrustLineEntry const& {
+            return poolShareTrustLine.current().data.trustLine();
+        };
+
+        auto poolID = poolTL().asset.liquidityPoolID();
+        auto pool = loadLiquidityPool(ltxInner, poolID);
+        auto balance = poolTL().balance;
+        auto cbSponsoringAccID = getTrustLineBacker(poolShareTrustLine);
+
+        // release reserves for pool share trustline that will be deleted
+        // later
+        {
+            auto trustAcc = stellar::loadAccount(ltxInner, accountID);
+            removeEntryWithPossibleSponsorship(ltxInner, ltxInner.loadHeader(),
+                                               poolShareTrustLine.current(),
+                                               trustAcc);
+
+            // using poolTL() after this will throw an exception
+            poolShareTrustLine.erase();
+        }
+
+        auto redeemIntoClaimableBalance =
+            [&ltxInner, &txSourceID, txSeqNum, opIndex, &poolID, &accountID,
+             &cbSponsoringAccID](Asset const& assetInPool,
+                                 int64_t amount) -> bool {
+            // if the claimant is the issuer, then we don't create the claimable
+            // balance
+            if (isIssuer(accountID, assetInPool))
+            {
+                return true;
+            }
+
+            // create a claimable balance for the asset in the pool
+            LedgerEntry claimableBalance;
+            claimableBalance.data.type(CLAIMABLE_BALANCE);
+
+            auto& claimableBalanceEntry =
+                claimableBalance.data.claimableBalance();
+
+            claimableBalanceEntry.balanceID =
+                getRevokeID(txSourceID, txSeqNum, opIndex, poolID, assetInPool);
+            claimableBalanceEntry.amount = amount;
+            claimableBalanceEntry.asset = assetInPool;
+            claimableBalanceEntry.claimants = {
+                makeUnconditionalClaimant(accountID)};
+
+            // if this asset isn't native
+            // 1. set clawback if it's set on the trustline
+            // 2. decrement liquidityPoolUseCount on the asset trustline
+            if (assetInPool.type() != ASSET_TYPE_NATIVE)
+            {
+                // trustline must exist
+                auto assetTrustLine =
+                    ltxInner.load(trustlineKey(accountID, assetInPool));
+
+                if (isClawbackEnabledOnTrustline(assetTrustLine))
+                {
+                    setClaimableBalanceClawbackEnabled(claimableBalanceEntry);
+                }
+
+                // Pool share trust line will be deleted, so do the required
+                // bookkeeping for the trust line now
+                if (--getTrustLineEntryExtensionV2(
```

Comment:
There's a bug here. We don't always reach this line (ex. when pool share trust line balance is 0).

Reply:
Fixed and added a test.